docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,165 @@
+# [Sparsity pattern and sparse matrices](@id topic-sparse-matrix)
+
+An important property of the finite element method is that it results in *sparse matrices*
+for the linear systems to be solved. On this page the topic of sparsity and sparse matrices
+are discussed.
+
+```@contents
+Pages = ["sparse_matrix.md"]
+Depth = 2:2
+```
+
+## Sparsity pattern
+
+The sparse structure of the linear system depends on many factors such as e.g. the weak
+form, the discretization, and the choice of interpolation(s). In the end it boils down to
+how the degrees of freedom (DoFs) *couple* with each other. The most common reason that two
+DoFs couple is because they belong to the same element. Note, however, that this is not
+guaranteed to result in a coupling since it depends on the specific weak form that is being
+discretized, see e.g. [Increasing the sparsity](@ref). Boundary conditions and constraints
+can also result in additional DoF couplings.
+
+If DoFs `i` and `j` couple, then the computed value in the eventual matrix will be
+*structurally nonzero*[^1]. In this case the entry `(i, j)` should be included in the
+sparsity pattern. Conversely, if DoFs `i` and `j` *don't* couple, then the computed value
+will be *zero*. In this case the entry `(i, j)` should *not* be included in the sparsity
+pattern since there is no need to allocate memory for entries that will be zero.
+
+The sparsity, i.e. the ratio of zero-entries to the total number of entries, is often *very*
+high and taking advantage of this results in huge savings in terms of memory. For example,
```

Comment:
Maybe we should be a bit more specific here to not mislead users accidentally.
```suggestion
The sparsity, i.e. the ratio of zero-entries to the total number of entries, is for most practical problems using low-order interpolations *very*
high for  and taking advantage of this results in huge savings in terms of memory. For example,
```
Note that this statement is not true anymore for e.g. high order Ansatz or interpolations spanning multiple elements.